x/projects/keeper/creation.go

```diff
@@ -2,58 +2,135 @@ package keeper
 	// the project with the same name already exists if no error has returned
 	if found {
-		return utils.LavaError(ctx, ctx.Logger(), "CreateEmptyProject_already_exist", map[string]string{"subscription": subscriptionAddress}, "project already exist for the current subscription with the same name")
+		return utils.LavaError(ctx, ctx.Logger(), "CreateEmptyProject_already_exist", map[string]string{"projectIndex": types.ProjectIndex(subscriptionAddress, projectData.GetName())}, "project already exist for the current subscription with the same name")
 	}
 
-	project.Policy.EpochCuLimit = cuPerEpoch
-	project.Policy.TotalCuLimit = totalCU
-	project.Policy.MaxProvidersToPair = providers
-	project.Policy.GeolocationProfile = geolocation
-
-	project.AppendKey(types.ProjectKey{Key: adminAddress, Types: []types.ProjectKey_KEY_TYPE{types.ProjectKey_ADMIN}, Vrfpk: vrfpk})
+	policy := projectData.GetPolicy()
+	if k.policyEmpty(policy) {
+		policy = types.Policy{
+			ChainPolicies:      []types.ChainPolicy{},
+			GeolocationProfile: math.MaxUint64,
+			TotalCuLimit:       plan.GetComputeUnits(),
+			EpochCuLimit:       plan.GetComputeUnitsPerEpoch(),
+			MaxProvidersToPair: plan.GetMaxProvidersToPair(),
+		}
+	}
 
-	err := k.RegisterDeveloperKey(ctx, adminAddress, project.Index, blockHeight, vrfpk)
+	err := k.ValidateChainPolicies(ctx, policy)
 	if err != nil {
 		return err
 	}
 
-	project.Enabled = enable
-	return k.projectsFS.AppendEntry(ctx, project.Index, blockHeight, &project)
-}
+	project.Policy = policy
 
-func (k Keeper) RegisterDeveloperKey(ctx sdk.Context, developerKey string, projectIndex string, blockHeight uint64, vrfpk string) error {
-	var developerData types.ProtoDeveloperData
-	_, found := k.developerKeysFS.FindEntry(ctx, developerKey, blockHeight, &developerData)
-	// a developer key with this address is not registered, add it to the developer keys list
-	if !found {
-		developerData.ProjectID = projectIndex
-		developerData.Vrfpk = vrfpk
-		err := k.developerKeysFS.AppendEntry(ctx, developerKey, blockHeight, &developerData)
+	for _, projectKey := range projectData.GetProjectKeys() {
+		err = k.RegisterKey(ctx, types.ProjectKey{Key: projectKey.GetKey(), Types: projectKey.GetTypes(), Vrfpk: projectKey.GetVrfpk()}, &project, blockHeight)
 		if err != nil {
 			return err
 		}
 	}
 
+	return k.projectsFS.AppendEntry(ctx, project.Index, blockHeight, &project)
+}
+
+func (k Keeper) policyEmpty(policy types.Policy) bool {
+	if len(policy.ChainPolicies) == 0 && policy.EpochCuLimit == uint64(0) && policy.GeolocationProfile == uint64(0) && policy.MaxProvidersToPair == 0 && policy.TotalCuLimit == 0 {
+		return true
+	}
+	return false
+}
+
+func (k Keeper) RegisterKey(ctx sdk.Context, key types.ProjectKey, project *types.Project, blockHeight uint64) error {
+	if project == nil {
+		return utils.LavaError(ctx, k.Logger(ctx), "RegisterKey_project_is_nil", nil, "project is nil")
+	}
+
+	for _, keyType := range key.GetTypes() {
```

Comment:
Let's simplify: ResgiterKey should expect/allow only one type. It will simplify the code (drop the unnecessary loop). If some code (project default?) needs to register a key with more than one type it call RegisterKey() twice, once with each role).
Also, can an admin key be registered in more than one project?  Can an admin key be also developer key except for the default project?  I thought no, and then the same tests as for developer key should apply.